examples/core/core_2d_camera_smooth_pixelperfect.c

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************************
+*
+*   raylib [core] example - smooth pixel-perfect camera
+*
+*   This example has been created using raylib 3.7 (www.raylib.com)
+*   raylib is licensed under an unmodified zlib/libpng license (View raylib.h for details)
+*
+*   Example contributed by Giancamillo Alessandroni ([discord]NotManyIdeas#9972 - [github]NotManyIdeasDev) and
+*   reviewed by Ramon Santamaria (@raysan5)
+*
+*   Copyright (c) 2021 Giancamillo Alessandroni (NotManyIdeas#9972) and Ramon Santamaria (@raysan5)
+*
+********************************************************************************************/
+
+#include "raylib.h"
+#include <math.h>
+
+int main(void)
+{
+    // Initialization
+    //--------------------------------------------------------------------------------------
+    const int screenWidth = 800;
+    const int screenHeight = 450;
+
+    const int virualScreenWidth = 160;
+    const int virtualScreenHeight = 90;
+
+    const float virtualRatio = (float)screenWidth / (float)virualScreenWidth;
+
+    InitWindow(screenWidth, screenHeight, "raylib [core] example - smooth pixel-perfect camera");
+
+    Camera2D worldSpaceCamera = { 0 }; // Game world camera
+    worldSpaceCamera.zoom = 1.0f;
+
+    Camera2D screenSpaceCamera = { 0 }; //Smoothing camera
+    screenSpaceCamera.zoom = 1.0f;
+
+    RenderTexture2D renderTexture = LoadRenderTexture(virualScreenWidth, virtualScreenHeight); //This is where we'll draw all our objects.
+
+    Rectangle firstRectangle = { 70.0f, 35.0f, 20.0f, 20.0f };
+    Rectangle secondRectangle = { 90.0f, 55.0f, 30.0f, 10.0f };
+    Rectangle thirdRectangle = { 80.0f, 65.0f, 15.0f, 25.0f };
+
+    //The renderTexture's height is flipped (in the source Rectangle), due to OpenGL reasons.
+    Rectangle renderTextureSource = { 0.0f, 0.0f, (float)renderTexture.texture.width, (float)-renderTexture.texture.height };
+    Rectangle renderTextureDest = { -virtualRatio, -virtualRatio, screenWidth + (virtualRatio * 2), screenHeight + (virtualRatio * 2) };
+
+    Vector2 origin = { 0.0f, 0.0f };
+
+    float rotation = 0.0f;
+    float degreesPerSecond = 60.0f;
+
+    float cameraX = 0.0f;
+    float cameraY = 0.0f;
+
+    SetTargetFPS(60);
+    //--------------------------------------------------------------------------------------
+
+    // Main game loop
+    while (!WindowShouldClose())    // Detect window close button or ESC key
+    {
+        // Update
+        //----------------------------------------------------------------------------------
+        rotation += degreesPerSecond * GetFrameTime(); // Rotate the rectangles.
+
+        // Make the camera move to demonstrate the effect.
+        cameraX = (sin(GetTime()) * 50.0f) - 10.0f;
+        cameraY = cos(GetTime()) * 30.0f;

```

Comment:
Internally raylib uses float versions of trigonometric functions. I'd suggest sticking to that for consistency.
```suggestion
        cameraX = (sinf(GetTime()) * 50.0f) - 10.0f;
        cameraY = cosf(GetTime()) * 30.0f;

```

Reply:
Changing it rn!